week11.tex

```diff
@@ -0,0 +1,172 @@
+\documentclass[11pt]{article}
+\usepackage[T1]{fontenc}
+\usepackage[utf8]{inputenc}
+\usepackage[english]{babel}
+\usepackage{amsmath,amssymb}
+\usepackage{url}
+
+\usepackage{pl-syntax/pl-judgments}
+
+\title{15-791 ATPL \\ Week 11 Notes \\ Dependent Types Part 1}
+\author{Jonathan Wilson}
+\date{\today}
+
+\begin{document}
+
+\maketitle{}
+
+\section*{Overview}
+
+This week begins our talks of dependent types. There is little focus in this week on logical relations or proofs and rather an emphasis on language design. Part of this is due to there not being one single canonical form of dependently typed language and part of it is due to many dependently typed languages allowing beta reduction and in the type system which quickly becomes undecidable (not to mention the performance costs).
+
+\section*{Review of Last week Call by push value}
+
+Last week we covered partiality as well as other effects in Call-by-Push-Value (CBPV). In order to adapt logical relations to partiality, we needed a lemma called compactness. In short, compactness says that if a program terminates, it does so in some finite number of steps. 
+
+Compactness: if force(susp$(x.C)) \mapsto^* ret(yes / no)$ then there exists an $n$ such that $force(susp^{(n)}(x.C)) \mapsto^* ret(yes / no)$
+
+There exists other ways to deal with partiality including Denotational semantics however this only pushes the problem away from the operational semantics without providing us any tools to handle effects or effect-ful control flow. For this course, we prefer to stay lower level and focus on actual evaluation with operational semantics.
+
+An alternative method that is a little more complicated but gives a little more insight into the problem is Pitts (spelling?). He uses a logical relation that is indexed by an upper bound on the number of steps that a program takes to terminate.
+
+Logical relations on partiality rely on fixed point induction. This gives rise to methods of interpreting types as "complete denotational partial orders".
```

Comment:
''

Reply:
Not sure what the change is. Is it untrue to say they are complete denotational partial orders? I have only heard of denotational from this one lecture.

Reply:
I think Tesla is recommending using
```
this kind of ``quotes in LaTeX''
```
instead of the `"` character, which renders weirdly.